### PR TITLE
MULE-17121: Upgrade maven-site-plugin to 3.8.2

### PR DESCRIPTION
**Pull request by lvzh90**

### CODE REVIEW

File: mule-extensions-xml-archetype/src/test/resources/projects/basic-with-spaces/reference/pom.xml [hunk 1]
```diff
@@ -26,8 +27,8 @@
         <maven.compiler.plugin.version>3.8.0</maven.compiler.plugin.version>
         <maven.install.plugin.version>2.5.2</maven.install.plugin.version>
         <maven.deploy.plugin.version>2.8.2</maven.deploy.plugin.version>
-        <maven.site.plugin.version>3.6.1</maven.site.plugin.version>
-        <maven.javadoc.plugin.version>3.0.1</maven.javadoc.plugin.version>
+        <maven.site.plugin.version>3.8.2</maven.site.plugin.version>
+        <maven.javadoc.plugin.version>3.1.0</maven.javadoc.plugin.version>
```
Comment (aiannucci) [inline mule-extensions-xml-archetype/src/test/resources/projects/basic-with-spaces/reference/pom.xml]: Use 3.1.1

File: mule-extensions-xml-archetype/src/test/resources/projects/basic-with-spaces/reference/pom.xml [hunk 2]
```diff
@@ -132,7 +133,19 @@
...
+                        <execution>
+                            <id>attach-javadocs</id>
+                            <goals>
+                                <goal>jar</goal>
+                            </goals>
+                        </execution>
+                    </executions>
+                    <configuration>
+                        <doclint>none</doclint>
+                        <additionalparam>${xDocLint}</additionalparam>
```
Comment (aiannucci) [inline mule-extensions-xml-archetype/src/test/resources/projects/basic-with-spaces/reference/pom.xml]: Do we need both `doclint` and `additionalparam` parameters?

Comment (lvzh90) [reply mule-extensions-xml-archetype/src/test/resources/projects/basic-with-spaces/reference/pom.xml]: We need only `doclint`

File: mule-extensions-xml-archetype/src/test/resources/projects/basic-with-spaces/reference/pom.xml (hunk 1 above)
Comment (lvzh90) [reply mule-extensions-xml-archetype/src/test/resources/projects/basic-with-spaces/reference/pom.xml]: Done

**Pull request merged**